ext/calendar/easter.c

```diff
@@ -48,11 +53,16 @@ static void _cal_easter(INTERNAL_FUNCTION_PARAMETERS, bool gm)
 		}
 	}
 
-	if (gm && (year<1970 || year>2037)) {				/* out of range for timestamps */
+	if (gm && (year<1970 || year>2037) && ZEND_LONG_MAX == INT32_MAX) { /* out of range for timestamps on 32bit systems */
 		zend_argument_value_error(1, "must be between 1970 and 2037 (inclusive)");
 		RETURN_THROWS();
 	}
```

Comment:
This  code should be #ifdef around a 32bit zend_long check to not compile this code at all, similarly for the 64 bit code (as the #else branch of this #ifdef)